Error that namespace was not found but namespace exists and test is able to run

```
ERRO[2024-06-28T09:28:01+02:00]k8s/k8s_namespace.go:53 Error getting namespace largenetwork-bigblock-8mib-latency-20240628-092801, err: error getting namespace largenetwork-bigblock-8mib-latency-20240628-092801: namespaces "largenetwork-bigblock-8mib-latency-20240628-092801" not found
```

--- a/pkg/k8s/k8s_namespace.go
+++ b/pkg/k8s/k8s_namespace.go
@@ -36,22 +36,20 @@ func (c *Client) DeleteNamespace(ctx context.Context, name string) error {
 }
 
 func (c *Client) GetNamespace(ctx context.Context, name string) (*corev1.Namespace, error) {
-	namespace, err := c.clientset.CoreV1().Namespaces().Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		return nil, ErrGettingNamespace.WithParams(name).Wrap(err)
-	}
-	return namespace, nil
+	return c.clientset.CoreV1().Namespaces().Get(ctx, name, metav1.GetOptions{})
 }
 
 func (c *Client) NamespaceExists(ctx context.Context, name string) bool {
 	_, err := c.GetNamespace(ctx, name)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			c.logger.Debugf("Namespace %s does not exist, err: %v", name, err)
-			return false
-		}
-		c.logger.Errorf("Error getting namespace %s, err: %v", name, err)
+	if err == nil {
+		return true
+	}
+
+	if errors.IsNotFound(err) {
+		c.logger.Debugf("Namespace %s does not exist, err: %v", name, err)
 		return false
 	}
-	return true
+
+	c.logger.Errorf("Error getting namespace %s, err: %v", name, err)
+	return false
 }

--- a/pkg/k8s/k8s_namespace.go
+++ b/pkg/k8s/k8s_namespace.go
@@ -4,7 +4,6 @@ import (
 	"context"
 
 	corev1 "k8s.io/api/core/v1"
-
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
@@ -43,14 +42,14 @@ func (c *Client) GetNamespace(ctx context.Context, name string) (*corev1.Namespa
 func (c *Client) NamespaceExists(ctx context.Context, name string) (bool, error) {
 	_, err := c.GetNamespace(ctx, name)
 	if err == nil {
-		return true
+		return true, nil
 	}
 
-	if errors.IsNotFound(err) {
+	if apierrs.IsNotFound(err) {
 		c.logger.Debugf("Namespace %s does not exist, err: %v", name, err)
-		return false
+		return false, nil
 	}
 
 	c.logger.Errorf("Error getting namespace %s, err: %v", name, err)
-	return false
+	return false, err
 }

--- a/pkg/k8s/k8s_namespace_test.go
+++ b/pkg/k8s/k8s_namespace_test.go
@@ -5,6 +5,7 @@ import (
 	"errors"
 
 	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes/fake"
@@ -126,11 +127,11 @@ func (s *TestSuite) TestDeleteNamespace() {
 
 func (s *TestSuite) TestGetNamespace() {
 	tests := []struct {
-		name        string
-		namespace   string
-		setupMock   func()
-		expectedErr error
-		expectedNS  *corev1.Namespace
+		name       string
+		namespace  string
+		setupMock  func()
+		assertErr  func(err error)
+		expectedNS *corev1.Namespace
 	}{
 		{
 			name:      "successful retrieval",
@@ -139,7 +140,9 @@ func (s *TestSuite) TestGetNamespace() {
 				err := s.createNamespace("existing-namespace")
 				s.Require().NoError(err)
 			},
-			expectedErr: nil,
+			assertErr: func(err error) {
+				s.Require().NoError(err)
+			},
 			expectedNS: &corev1.Namespace{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "existing-namespace",
@@ -150,14 +153,12 @@ func (s *TestSuite) TestGetNamespace() {
 			name:      "namespace not found",
 			namespace: "non-existent-namespace",
 			setupMock: func() {
-				s.client.Clientset().(*fake.Clientset).
-					PrependReactor("get", "namespaces",
-						func(action k8stesting.Action) (handled bool, ret runtime.Object, err error) {
-							return true, nil, errors.New("namespaces \"non-existent-namespace\" not found")
-						})
+				// no need to mock
 			},
-			expectedErr: k8s.ErrGettingNamespace.WithParams("non-existent-namespace").
-				Wrap(errors.New("namespaces \"non-existent-namespace\" not found")),
+			assertErr: func(err error) {
+				s.Require().Error(err)
+				s.Assert().True(apierrs.IsNotFound(err))
+			},
 			expectedNS: nil,
 		},
 		{
@@ -170,8 +171,10 @@ func (s *TestSuite) TestGetNamespace() {
 							return true, nil, errors.New("internal server error")
 						})
 			},
-			expectedErr: k8s.ErrGettingNamespace.WithParams("error-namespace").
-				Wrap(errors.New("internal server error")),
+			assertErr: func(err error) {
+				s.Require().Error(err)
+				s.Assert().Equal(err.Error(), "internal server error")
+			},
 			expectedNS: nil,
 		},
 	}
@@ -181,13 +184,7 @@ func (s *TestSuite) TestGetNamespace() {
 			tt.setupMock()
 
 			ns, err := s.client.GetNamespace(context.Background(), tt.namespace)
-			if tt.expectedErr != nil {
-				s.Require().Error(err)
-				s.Assert().ErrorIs(err, tt.expectedErr)
-				return
-			}
-
-			s.Require().NoError(err)
+			tt.assertErr(err)
 			s.Assert().EqualValues(tt.expectedNS, ns)
 		})
 	}

--- a/pkg/traefik/traefik.go
+++ b/pkg/traefik/traefik.go
@@ -162,7 +162,7 @@ func (t *Traefik) Deploy(ctx context.Context) error {
 			},
 		},
 	}
-	_, err = t.K8s.Clientset().AppsV1().Deployments(t.K8s.Namespace()).
+	_, err = t.K8sClient.Clientset().AppsV1().Deployments(t.K8sClient.Namespace()).
 		Create(ctx, traefikDeployment, metav1.CreateOptions{})
 	if err != nil {
 		return ErrTraefikDeploymentCreationFailed.Wrap(err)
@@ -283,7 +283,7 @@ func (t *Traefik) createMiddleware(ctx context.Context, serviceName, middlewareN
 		Resource: "middlewares",
 	}
 
-	_, err := t.K8s.DynamicClient().Resource(middlewareResource).Namespace(t.K8s.Namespace()).
+	_, err := t.K8sClient.DynamicClient().Resource(middlewareResource).Namespace(t.K8sClient.Namespace()).
 		Create(ctx, middleware, metav1.CreateOptions{})
 	if err != nil {
 		return ErrTraefikMiddlewareCreationFailed.Wrap(err)
@@ -339,7 +339,7 @@ func (t *Traefik) createIngressRoute(
 		},
 	}
 
-	_, err = t.K8s.DynamicClient().Resource(ingressRouteGVR).Namespace(t.K8s.Namespace()).
+	_, err = t.K8sClient.DynamicClient().Resource(ingressRouteGVR).Namespace(t.K8sClient.Namespace()).
 		Create(ctx, ingressRoute, metav1.CreateOptions{})
 	if err != nil {
 		return ErrTraefikIngressRouteCreationFailed.Wrap(err)

